llvm/utils/TableGen/DXILEmitter.cpp

```diff
@@ -268,36 +296,45 @@ static void emitDXILIntrinsicMap(std::vector<DXILOperationDesc> &Ops,
 
 static std::string emitDXILOperationFnAttr(StringRef FnAttr) {
   return StringSwitch<std::string>(FnAttr)
-      .Case("rn", "Attribute::ReadNone")
-      .Case("ro", "Attribute::ReadOnly")
+      .Case("NoReadMemory", "Attribute::ReadNone")
+      .Case("ReadMemory", "Attribute::ReadOnly")
```

Comment:
Why not just ReadNone and ReadOnly?

Reply:
> Why not just ReadNone and ReadOnly?

The thought was to emphasize *memory* read, but I am open to changing if the terms chosen don't sound palatable.

Reply:
ReadMemory lost the only part of ReadOnly, but ReadOnlyMemory feels bad too :(
Maybe OnlyReadMemory?

Reply:
> ReadMemory lost the only part of ReadOnly, but ReadOnlyMemory feels bad too :( Maybe OnlyReadMemory?

How about `MemoryReadNone` and `MemoryReadOnly`?

Reply:
Don't these map directly to the llvm function attributes? If so I think we should match those names (`readonly` and `readnone`).

Reply:
> Don't these map directly to the llvm function attributes? 

Yes. 

> If so I think we should match those names (`readonly` and `readnone`). Using the decided naming convention (e.g. CamlCase).

As the `DXIL.td` is expected to be the eventual single source of reference for DXIL operations and affords declarative specification, the thought was to be a bit more descriptive than C++ sources, when possible.

I can change them as suggested.

Reply:
I changed the type of `Attribute` to `EnumAttr` type to get the benefit of appropriately constrained values that can be specified and validating at compile-time as opposed to using a `string`.